modules/diskq/qdisk.c

```diff
@@ -421,6 +421,12 @@ qdisk_push_tail(QDisk *self, GString *record)
   return TRUE;
 }
 
+static gboolean
+_is_trying_to_read_from_unwritten_space(QDisk *self)
```

Comment:
I agree with @Kokan.

In the meantime, if this patch remains: Move the other refactor here: https://github.com/syslog-ng/syslog-ng/pull/3743/commits/3f992d88ac267f35b42a74ab0128180e633bf46c#diff-3482ded9f3982932047654ee373fc4530b82f6fa47566735910ec9b5cff3957aL531

Reply:
I'd just drop this and use `self->hdr->read_head == self->hdr->write_head` directly, as it is eloquent enough.
Or we can just rename this to `_is_empty()`, but that's confusing because we have a similar method with a different implementation (it is based on `length`).

@alltilla Can we remove this one?

Reply:
Sure! :)

Reply:
Removed.